### PR TITLE
Submission relationship lookup externalsources: Missing i18n labels & other issues

### PR DESCRIPTION
**Pull request by MarieVerdonck**
## References
https://github.com/DSpace/dspace-angular/issues/867

## Description
This PR adds the missing labels mentioned in https://github.com/DSpace/dspace-angular/issues/867 .

It also solves an issue that the search result list items for the external sources (ex PubMed) were not rendering in the lookup Author modal during submission. Cause was that there was no @listableObjectComponent for the case that there were *no* name variants configured.

  Lastly, it solves an issue that the count of number of results in sources tab wasn’t updating (remained on (0)) for the external sources.

## Instructions for Reviewers
Submission => Author lookup => External source tab like PubMed/arXiv

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/assets/i18n/en.json5 [hunk 1]
```diff
@@ -2960,11 +2960,11 @@
 
   "submission.sections.describe.relationship-lookup.external-source.import-button-title.Journal Volume": "Import remote journal volume",
```
Comment (tdonohue) [inline src/assets/i18n/en.json5:2961]: If I hover over the import buttons for `PubMed` or `arXiv`, I see a missing key of `submission.sections.describe.relationship-lookup.external-source.import-button-title.isAuthorOfPublication`

**Review by tdonohue — CHANGES_REQUESTED** [review]
@MarieVerdonck : This looks good overall and mostly works.  I just found one additional missing i18n key (see inline below).

As a sidenote, in testing this PR, I've noticed that the Author "Lookup" button really should NOT be allowing you to import from many of these external sources, as some of these external sources are NOT authors.  Namely, ONLY "Local Authors", ORCID and LC Names should be included.  I'm not sure why we even allow searching Sherpa Journals, Sherpa Publishers, PubMed or arXiv from an Author lookup button, as those are obviously different types of Entities.  However, this is not a bug in this PR, so I could log it as a separate ticket if you and @artlowel agree.

Overall though, this code looks good & it works. Just one i18n key needs to be added and then it's ready to merge.

File: src/assets/i18n/en.json5 (hunk 1 above)
Comment (MarieVerdonck) [reply src/assets/i18n/en.json5:2961]: @tdonohue Added it in

**Comment by artlowel** [comment]
Yes that should be a separate ticket @tdonohue. Currently there is no way to configure the available sources specifically for each field. It'll likely require both back-end and front-end changes

**Review by tdonohue — APPROVED** [review]
👍 Looks good now.  Thanks @MarieVerdonck !

(As a sidenote, I've moved the reported issue above to a new ticket...see #943 )

**Pull request merged**